config/initializers/omniauth.rb

```diff
@@ -40,13 +40,22 @@ def callback_phase
         fail!(:timeout, e)
       rescue ::SocketError => e
         fail!(:failed_to_connect, e)
+      # Adding this provides some nicer / additional logging but not why the error was caused
+      rescue Rack::OAuth2::Client::Error => e
+        fail!(:invalid_client, e)
```

Comment:
Could we provide additional text saying something like:

> Could not perform OAuth authentication. Do you have 'dfe_signin.secret' set in Settings?

And we could expand on this as we find other common causes for this problem.